tests/rptest/tests/audit_log_tests.py

```diff
@@ -0,0 +1,152 @@
+                self.logger.info(f"Ingested: {len(new_records)} records")
+                self.records += [r for r in new_records if self.filter_fn(r)]
+
+            def is_finished(self):
+                return stop_cond(self.records)
+
+        mapper = MessageMapper(self.redpanda.logger, filter_fn, stop_cond)
+        n = self._audit_log_total_number_records()
+
+        consumer = RpkConsumer(self._ctx, self.redpanda, self.audit_log)
+        consumer.start()
+
+        def predicate():
+            mapper.ingest(consumer.messages)
+            return consumer.message_count >= n and mapper.is_finished()
+
+        wait_until(predicate, timeout_sec=timeout_sec, backoff_sec=backoff_sec)
+        consumer.stop()
+        consumer.free()
+        return mapper.records
+
+    @cluster(num_nodes=4)
+    def test_audit_log_functioning(self):
+        """
+        Ensures that the audit log can be produced to when the audit_enabled()
+        configuration option is set, and that the same actions do nothing
+        when the option is unset. Furthermore verifies that the internal duplicate
+        aggregation feature is working.
+        """
+        def is_api_match(matches, record):
+            regex = re.compile(
+                "http:\/\/(?P<address>.*):(?P<port>\d+)\/v1\/(?P<handler>.*)")
+            match = regex.match(record['api']['operation'])
+            if match is None:
+                raise RuntimeError('Record out of spec')
+            return match.group('handler') in matches
+
+        def number_of_records_matching(filter_by, n_expected):
+            filter_fn = partial(is_api_match, filter_by)
+
+            def aggregate_count(records):
+                # Duplicate records are combined with the 'count' field
+                def combine(acc, x):
+                    return acc + (1 if 'count' not in x else x['count'])
+
+                return reduce(combine, records, 0)
+
+            stop_cond = lambda records: aggregate_count(records) == n_expected
+            return self._read_all_from_audit_log(filter_fn, stop_cond)
+
+        self._wait_for_audit_log(timeout_sec=10)
+
+        # The test override the default event type to 'heartbeat', therefore
+        # any actions on the admin server should not result in audit msgs
+        api_calls = {
+            'features/license': self.admin.get_license,
+            'cluster/health_overview': self.admin.get_cluster_health_overview
+        }
+        api_keys = api_calls.keys()
+        call_apis = lambda: [fn() for fn in api_calls.values()]
+        for _ in range(0, 500):
+            call_apis()
+
+        # Wait 1 second because asserting that 0 records of a type occur
+        # will not continuously loop for the records we are expected not to
+        # see. 1 second is long enough for the audit fibers to run and push
+        # data to the audit topic.
+        time.sleep(1)
+        _ = number_of_records_matching(api_keys, 0)
```

Comment:
I think this test should be moved to the end after you have verified that call_apis lands data into the audit log. Then you disable audit log, do the same thing, and verify that the audit log doesn't grow. And wait longer than 1 second that isn't reliable amount of time for nearly anything. Better would be to test the disabled case using metrics, but the stakes are pretty low since the biggest qualifier is that audit logging works when _enabled_.